src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -786,4 +786,41 @@
       it { is_expected.to redirect_to(action: :show, project: user.home_project) }
     end
   end
+
+  describe "POST #add_maintained_project" do
+    before do
+      login user
+    end
+
+    context "maintained project successfully added" do
+      before do
+        user.home_project.update(kind: 'maintenance')
+      end
+
+      context "maintained project successfully added" do
+        before do
+          post :add_maintained_project, project: user.home_project, maintained_project: user.home_project.name
+        end
+
+        it { expect(flash[:notice]).to eq("Added #{user.home_project.name} to maintenance") }
+        it { is_expected.to redirect_to(action: 'maintained_projects', project: user.home_project) }
+      end
+
+      # raise the exception in the before_filter set_maintained_project
+      it "#add_maintained_project raise excepction with invalid maintained project" do
+        expect {
+          post :add_maintained_project, project: user.home_project, maintained_project: "invalid"
+        }.to raise_exception ActiveRecord::RecordNotFound
+      end
+    end
+
+    # redirect in the before_filter require_maintenance_project
+    context "#add_maintained_project fails without maintenance kind for a valid maintained project" do
```

Comment:
But I think thar it is not obvious in this case neither beause the variable that is checked in the `require_maintenance_project` is set in another before filter and if that test fails this comment will be really helpful to be aware of that.

Also, for coherence as a comment like this is in the `remove_maintained_project` test which is really similar.